### ISSUE TITLE
First thank you for your code. May I have a question?

### ISSUE THREAD

**Issue by Williamyichang** [dhkdnduq/PatchCore_anomaly_detection#11]
In your code, I didn't find any implement for below formula, Maybe  my eyes too small.
![image](https://user-images.githubusercontent.com/42769951/175757154-18b948f8-b20a-4590-9ae4-379aa1300cd5.png)
![image](https://user-images.githubusercontent.com/42769951/175757168-aa89b8d6-ac50-4f68-8854-375c08c53094.png)

**Comment by dhkdnduq** [comment]
When training, the maximum and minimum values of all images are obtained and stored.
It is used for thresholds for determining abnormalities or create  heat map

##
double max_score = cfg_.vanomaly[category].anomalyMaxScore;
double min_score = cfg_.vanomaly[category].anomalyMinScore;
##

**Comment by Williamyichang** [comment]
> When training, the maximum mean and minimum mean values of all images are obtained and stored. It is used for thresholds for determining abnormalities or create heat map
> 
> Look at this. def analyze_data(self):
> 
> double max_score = cfg_.vanomaly[category].anomalyMaxScore; double min_score = cfg_.vanomaly[category].anomalyMinScore;

Thank you for your reply, May I have one more questions?  Is the code "scores = (score_pathces - min_score) / (max_score - min_score)" is equal to below formula?
![image](https://user-images.githubusercontent.com/42769951/176462980-19155065-4801-4798-aa2b-fa2455f05087.png)

**Comment by dhkdnduq** [comment]
> Thank you for your reply, May I have one more questions? Is the code "scores = (score_pathces - min_score) / (max_score - min_score)" is equal to below formula?

No.It's for the heat map.

There is no GT in the industry.
Therefore, the average value of the patch scores was used.

score_patches = knn(torch.from_numpy(embedding_test).cuda())[0].cpu().detach().numpy()
np.mean(score_patches )  *Require precise processing of data depending on the situation

Then, find the threshold value between OK and NG data through "Find_Optimal_Cutoff."

**Comment by Williamyichang** [comment]
> 

Thank you for the explanation. I need take time to digest.
